### PR TITLE
fix(llm): make logprobs opt-in (default off)

### LINKED ISSUE #49: bug: openai-chat provider sends logprobs unconditionally → HTTP 400 on Gemini/Ollama
## Bug

`OpenAICompatProvider.create()` adds `"logprobs": True` to **every** chat-completions request, with a comment assuming unsupported endpoints "ignore" the field. Several don't — they reject it:

- **Gemini** (`https://generativelanguage.googleapis.com/v1beta/openai/`): `400 Invalid JSON payload received. Unknown name "logprobs": Cannot find field.`
- **Ollama**: rejects `logprobs` in many versions.

Arbor's planner call fails on turn 1, so it stays in intake and never launches a run:

```
LLM call failed (turn 1, attempt 1/3): Error code: 400 - ... Unknown name "logprobs" ...
╭─ model unavailable ─────────────────────────────────────╮
│ The planning model did not answer, so Arbor stayed in   │
│ intake instead of launching a run.                      │
╰─────────────────────────────────────────────────────────╯
```

## Root cause

`src/core/llm/openai_compat.py`, `OpenAICompatProvider.create()`:

```python
params: dict[str, Any] = {
    "model": self.model,
    "messages": oai_messages,
    "max_tokens": max_tokens,
    # Token-faithful traces: ask for sampled-token logprobs. Endpoints that
    # don't support it ignore these; we read back None and degrade cleanly.
    "logprobs": True,          # ← unconditional
}
```

The "endpoints ignore it" assumption is false for Gemini and Ollama. The `litellm` backend inherits the same `create()`, and `litellm.drop_params = True` does **not** help — litellm considers `logprobs` a *supported* param for `gemini/*`, so it isn't stripped (verified: the native Gemini API returns a separate `Logprobs is not enabled for models/gemini-2.5-flash` 400).

## Repro

```
arbor config init --provider openai-chat \
  --model gemini-2.5-flash \
  --base-url https://generativelanguage.googleapis.com/v1beta/openai/ \
  --api-key <key> --force
arbor   # type anything → "model unavailable"
```

A plain call **without** `logprobs` to the same endpoint succeeds, confirming the field is the sole blocker.

## Proposed fix

Make `logprobs` an **opt-in** `llm.logprobs` config field (default `false`), plumbed through `create_provider` to both `OpenAICompatProvider` and `LiteLLMProvider`. A generic OpenAI-compatible provider cannot assume the endpoint supports logprobs, so off-by-default makes every listed backend (DeepSeek / Qwen / GLM / vLLM / Ollama / Gemini) work out of the box. Users who need token-faithful trajectory traces (#41) set `llm.logprobs: true` (supported on OpenAI / vLLM / DeepSeek).

`_extract_logprobs` already degrades to `None` when absent, so traces simply skip the field when off — no behavioral loss beyond the optional SFT capture.

I have a PR ready: `fix(llm): make logprobs opt-in (default off)`.

### PR DESCRIPTION
**Pull request by redoracle**
## Summary

`OpenAICompatProvider.create()` sent `"logprobs": True` on every chat-completions call, assuming unsupported endpoints ignore the field. They don't — Gemini's OpenAI-compat layer and Ollama reject it with HTTP 400, leaving Arbor stuck in intake ("model unavailable") on turn 1. The `litellm` backend inherits the same `create()` and `litellm.drop_params` doesn't help (litellm considers `logprobs` a supported param for `gemini/*`, so it isn't stripped).

This gates `logprobs` behind a new **opt-in** `llm.logprobs` config field (default `false`), plumbed through `create_provider` to both `OpenAICompatProvider` and `LiteLLMProvider`. A generic OpenAI-compatible provider cannot assume the endpoint supports logprobs, so off-by-default makes every listed backend (DeepSeek / Qwen / GLM / vLLM / Ollama / Gemini) work out of the box. `_extract_logprobs` already degrades to `None`, so token-faithful traces (#41) simply skip when off — users who need them set `llm.logprobs: true` (OpenAI / vLLM / DeepSeek).

## Linked issue

Closes #49

## Type of change

- [x] 🐞 Bug fix (`fix`)

## How was this tested?

```text
$ python tests/test_openai_compat_logprobs.py
ok  test_create_provider_plumbs_logprobs
ok  test_litellm_provider_inherits_gate
ok  test_logprobs_defaults_off_in_config
ok  test_logprobs_not_sent_by_default
ok  test_logprobs_sent_when_enabled
5 passed

$ python -m pytest tests/ -q
........ [full suite green, 3 pre-existing skips]
```

End-to-end against the real Gemini `/v1beta/openai/` endpoint with the cloned source:
- default config (`logprobs` off) → `OK`, `logprobs=None` ✅
- `logprobs=true` → reproduces the exact `400 Unknown name "logprobs"` ✅ (gate is real)

## Checklist

- [x] PR title follows Conventional Commits (`fix(llm): ...`).
- [x] The change is focused and reasonably small (9 files, +136/−3).
- [x] I ran the relevant tests locally and they pass.
- [x] Docs / examples updated (`docs/configuration.md`, `docs/configuration.zh.md`, `examples/research_config.example.yaml`). The READMEs don't enumerate per-field `llm:` options (they point to `docs/configuration.md`), so no README change was needed.
- [x] No secrets, API keys, or tokens are included in the diff.

### CODE REVIEW

**Comment by redoracle** [comment]
### Why default `logprobs: false` is the correct choice 

A natural question on this PR: *isn't `logprobs` needed for Arbor's core?* I traced the codebase and the answer is **no** — and the project's own docs already specify the opt-in design this PR implements.

**1. No core logic reads `logprobs`.** Grepping `src/coordinator/`, `src/core/agent.py`, `src/core/context.py` for `.logprobs` returns nothing. The only consumer is `src/trajectory.py:142`:

```python
"logprobs": getattr(response, "logprobs", None),  # None unless provider gives it
```

It captures the field *if present* for the SFT dump. Planning, ideation, experiment execution, eval, backpropagation, and reporting never look at it. With `logprobs=None` (the new default), Arbor's research loop is identical.

**2. The docs already mark token-faithful traces as opt-in and OpenAI-only.** From `docs/dev/trajectory-export.md`:

> - **Coarse trajectory** (`trajectory.jsonl`): *"Cheap, always on."* — no logprobs needed; this is what feeds SFT by default.
> - **Token-faithful trace** (`tokens.jsonl`, **opt-in**): *"messages, sampled token_ids, logprobs, loss mask… RL-ready, Polar-compatible. **Needs a provider that returns logprobs/token_ids (OpenAI yes; Anthropic limited…)**"*

So the documented design is: token-faithful (the logprobs feature) is **opt-in** and **provider-dependent (OpenAI yes)**. But the code was sending `logprobs=True` **unconditionally on every endpoint** — contradicting its own documentation, and 400-ing on Gemini/Ollama. This PR aligns the implementation with the documented intent.

**3. The README doesn't mention logprobs at all.** Every advertised capability (hypothesis-tree search, dev/test eval, worktree isolation, guarded merge, alphaXiv search, multi-backend, steerability) is independent of it. "Self-evolution" in the README means experience reuse (recalling past findings at intake), not fine-tuning.

**Net:** the unconditional send broke 5 of the 6 backends listed in `OpenAICompatProvider`'s docstring (DeepSeek/Qwen/GLM/vLLM/Ollama + Gemini via litellm) for a feature that is documented as opt-in, works only on OpenAI, and has no core consumer. Default-off makes every listed backend work out of the box; users who need RL-ready token traces set `llm.logprobs: true` on OpenAI/vLLM/DeepSeek — exactly as `docs/dev/trajectory-export.md` already prescribes.

**Comment by redoracle** [comment]
Withdrawing for now — the work continues on the redoracle fork (https://github.com/redoracle/Arbor, branch `redoracle`). Closing to keep the upstream tracker tidy; happy to reopen if useful.
